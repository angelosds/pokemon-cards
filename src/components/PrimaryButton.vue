<template>
  <button class="primary-button" :type="type" @click="onClick">
    <slot />
  </button>
</template>

<script lang="ts">
import { Component, Emit, Prop, Vue } from "vue-property-decorator";

@Component
export default class PrimaryButton extends Vue {
  @Prop({ default: "button" }) private readonly type?: string;

  @Emit()
  onClick(event: Event) {
    return event;
  }
}
</script>

<style lang="scss" scoped>
.primary-button {
  background-color: $primary-color;
  border-radius: 8px;
  border: 0;
  color: $text-color-inverse;
  font-weight: 700;
  outline: 0;
  padding: 12px 24px;

  &:hover,
  &:focus {
    background-color: $primary-feedback-color;
  }

  &:focus {
    box-shadow: inset 0 0 0 4px $primary-color;
  }
}
</style>
