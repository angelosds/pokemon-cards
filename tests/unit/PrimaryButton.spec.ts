import { shallowMount } from "@vue/test-utils";

import PrimaryButton from "@/components/PrimaryButton.vue";

describe("PrimaryButton.vue", () => {
  const build = (propsData?: object, slots?: object) => {
    const wrapper = shallowMount(PrimaryButton, {
      propsData: { ...propsData },
      slots: { ...slots }
    });

    return { wrapper };
  };

  it("renders the component", () => {
    const { wrapper } = build();

    expect(wrapper).toMatchSnapshot();
  });

  it("renders text in a slot", () => {
    const defaultSlotText = "default slot text";
    const { wrapper } = build(undefined, { default: defaultSlotText });

    expect(wrapper.text()).toMatch(defaultSlotText);
  });

  it("sets a button default type", () => {
    const { wrapper } = build();

    expect(wrapper.attributes("type")).toEqual("button");
  });

  it("sets a button custom type", () => {
    const type = "submit";
    const { wrapper } = build({ type });

    expect(wrapper.attributes("type")).toEqual(type);
  });

  it("emits a onClick event", () => {
    const { wrapper } = build();

    wrapper.trigger("click");

    expect(wrapper.emitted("on-click")).toBeTruthy();
  });
});
