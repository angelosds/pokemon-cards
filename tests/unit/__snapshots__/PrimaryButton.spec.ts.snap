// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton.vue renders the component 1`] = `<button type="button" class="primary-button"></button>`;
